### PR TITLE
1903 : renamed functions to remove ambiguity

### PR DESCRIPTION
**Pull request by ajnovice**
I have renamed some functions to remove the confusion about their functionality. [1903](https://github.com/OpenMined/PySyft/issues/1903)

### CODE REVIEW

File: syft/frameworks/torch/hook/hook.py [hunk 1]
```diff
@@ -540,7 +540,7 @@ def overloaded_attr(self, *args, **kwargs):
             """
 
             # Replace all syft tensor with their child attribute
-            new_self, new_args, new_kwargs = syft.frameworks.torch.hook_args.hook_method_args(
+            new_self, new_args, new_kwargs = syft.frameworks.torch.hook_args.hook_object_method_with_args(
```
Comment (LaRiffle) [inline syft/frameworks/torch/hook/hook.py]: Maybe `unwrap_args_in_method` would be more accurate (or `hook_args_in_method` but maybe less clear)

File: syft/frameworks/torch/hook/hook_args.py [hunk 2]
```diff
@@ -122,7 +122,7 @@ def hook_method_args(attr, method_self, args, kwargs):
         new_self, new_args = hook_args((method_self, args))
 
     except (IndexError, KeyError, AssertionError):  # Update the function in case of an error
-        args_hook_function, _ = build_hook_args_function((method_self, args))
+        args_hook_function, _ = build_hook_for_args_with_function((method_self, args))
```
Comment (LaRiffle) [inline syft/frameworks/torch/hook/hook_args.py]: same here maybe unwrap_args_in_function is more precise

File: syft/frameworks/torch/hook/hook_args.py [hunk 3]
```diff
@@ -254,7 +256,7 @@ def hook_response(attr, response, wrap_type, wrap_args={}, new_self=None):
     return new_response
 
 
-def build_hook_response_function(response, wrap_type, wrap_args):
+def build_hook_for_response_with_function(response, wrap_type, wrap_args):
```
Comment (LaRiffle) [inline syft/frameworks/torch/hook/hook_args.py]: maybe `wrap_reponse_in_function`?

Comment (LaRiffle) [reply syft/frameworks/torch/hook/hook_args.py]: with the build_ prefix sorry

File: syft/frameworks/torch/hook/hook.py (hunk 1 above)
Comment (ajnovice) [reply syft/frameworks/torch/hook/hook.py]: `hook_method_args` function was hooking the method of object with args so I named it as `hook_object_method_with_args` in layman terms. `unwrap_args_in_method` would still be a bit confusing about the its functionality. If you want, we could name it `unwrap_args_with_method`

File: syft/frameworks/torch/hook/hook_args.py (hunk 2 above)
Comment (ajnovice) [reply syft/frameworks/torch/hook/hook_args.py]: `unwrap_args_with_function`  maybe more precise.

File: syft/frameworks/torch/hook/hook_args.py (hunk 3 above)
Comment (ajnovice) [reply syft/frameworks/torch/hook/hook_args.py]: maybe `build_wrap_reponse_with_function` for `build_hook_response_function`?

File: syft/frameworks/torch/hook/hook_args.py [hunk 4]
```diff
@@ -131,8 +131,8 @@ def hook_method_args(attr, method_self, args, kwargs):
     return new_self, new_args, kwargs
 
 
-def hook_function_args(attr, args, kwargs, return_args_type=False):
-    """See hook_method_args for details
+def hook_function_with_args(attr, args, kwargs, return_args_type=False):
```
Comment (LaRiffle) [inline syft/frameworks/torch/hook/hook_args.py]: Same renaming should apply than:
hook_method_args -> unwrap_args_with_method
hook_function_args -> unwrap_args_with_function

File: syft/frameworks/torch/hook/hook_args.py [hunk 5]
```diff
@@ -174,7 +174,7 @@ def hook_function_args(attr, args, kwargs, return_args_type=False):
         return new_args, kwargs, new_type
 
 
-def build_hook_args_function(args, return_tuple=False):
+def unwrap_args_with_function(args, return_tuple=False):
```
Comment (LaRiffle) [inline syft/frameworks/torch/hook/hook_args.py]: Maybe my last review was confusing, I was expecting:
hook_function_args -> unwrap_args_from_function
so for build_hook_args_function we should do:
build_hook_args_function -> build_unwrap_args_from_function

File: syft/frameworks/torch/hook/hook_args.py [hunk 6]
```diff
@@ -300,7 +300,7 @@ def build_rule(args):
         return 0
 
 
-def build_args_hook(args, rules, return_tuple=False):
+def build_hook_for_args_child_with_rules(args, rules, return_tuple=False):
```
Comment (LaRiffle) [inline syft/frameworks/torch/hook/hook_args.py]: Maybe avoid "hook"
build_args_hook -> build_unwrap_args_with_rules

File: syft/frameworks/torch/hook/hook_args.py [hunk 7]
```diff
@@ -448,7 +450,7 @@ def four_layers(idx1, *ids):
 get_element_at = {1: one_layer, 2: two_layers, 3: three_layers, 4: four_layers}
 
 
-def build_response_hook(response, rules, wrap_type, wrap_args, return_tuple=False):
+def build_response_hook_with_rule(response, rules, wrap_type, wrap_args, return_tuple=False):
```
Comment (LaRiffle) [inline syft/frameworks/torch/hook/hook_args.py]: Maybe avoid "hook"
build_response_hook -> build_wrap_response_with_rules

File: syft/frameworks/torch/hook/hook_args.py (hunk 4 above)
Comment (LaRiffle) [reply syft/frameworks/torch/hook/hook_args.py]: Even better: hook_function_args -> unwrap_args_from_function

File: syft/frameworks/torch/hook/hook_args.py [hunk 8]
```diff
@@ -89,7 +89,7 @@
 ambiguous_functions = {"torch.unbind", "unbind", "torch.stack", "stack", "torch.mean", "torch.sum"}
 
 
-def hook_method_args(attr, method_self, args, kwargs):
+def unwrap_args_with_method(attr, method_self, args, kwargs):
```
Comment (LaRiffle) [inline syft/frameworks/torch/hook/hook_args.py]: Actually maybe use 
hook_method_args -> unwrap_args_from_method
to improve clarity

File: syft/frameworks/torch/hook/hook_args.py [hunk 9]
```diff
@@ -254,7 +254,7 @@ def hook_response(attr, response, wrap_type, wrap_args={}, new_self=None):
     return new_response
 
 
-def build_hook_response_function(response, wrap_type, wrap_args):
+def build_wrap_reponse_with_function(response, wrap_type, wrap_args):
```
Comment (LaRiffle) [inline syft/frameworks/torch/hook/hook_args.py]: build_wrap_reponse_with_function -> build_wrap_reponse_from_function
